### PR TITLE
PCA - Fixed PCA integration test to be parameterized on artifact updated event objects

### PR DESCRIPTION
**Pull request by chamilad**
This fix enables to test the authenticated git clone on PCA and also fixes minor issues in test structure.

### CODE REVIEW

**Comment by imesh** [comment]
Thanks for the modification @chamilad! I have now merged this to master branch.

**Comment by chamilad** [comment]
Thanks @imesh !

File: components/org.apache.stratos.python.cartridge.agent/src/test/java/org/apache/stratos/python.cartridge.agent/test/PythonCartridgeAgentTest.java [hunk 1]
```diff
@@ -104,97 +170,118 @@ public static void tearDown() {
...
+                                // Simulate server socket
+                                startServerSocket(9080);
+                            }
+                            if (line.contains("Artifact repository found")) {
+                                // Send artifact updated event
+                                publishEvent(artifactUpdatedEvent);
+                            }
+
+                            if (line.contains("Exception in thread") || line.contains("ERROR")) {
+                                //throw new RuntimeException(line);
```
Comment (prabathabey) [inline components/org.apache.stratos.python.cartridge.agent/src/test/java/org/apache/stratos/python.cartridge.agent/test/PythonCartridgeAgentTest.java]: Any particular reason why the body of the if condition is left blank? Shouldn't this be logged or handled as how it's already done in other cases?

Comment (chamilad) [inline components/org.apache.stratos.python.cartridge.agent/src/test/java/org/apache/stratos/python.cartridge.agent/test/PythonCartridgeAgentTest.java]: Hi @prabathabey , yes at the moment this does not serve any purpose. However the plan is to check the class of the error thrown and fail the test accordingly. This should be commented out.

**Pull request closed without merging**